connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/ProcessingContext.java

```diff
@@ -216,4 +216,10 @@ public void reporters(Collection<ErrorReporter> reporters) {
         this.reporters = reporters;
     }
 
+    @Override
+    public void close() throws Exception {
+        for (ErrorReporter reporter : reporters) {
+            reporter.close();
```

Comment:
The exception caused by reporter will break this loop. Should we keep the exception and then throw it after all reporters are closed?

Reply:
Yes, thanks for catching this oversight. I've added logic to suppress the exceptions from each of the reporters, but i'm not sure if that's the idiomatic way to combine multiple exceptions. I've also added a test to verify that this error collection behavior is present.